chore(config): Removes several $CONFIG keys and deprecated APIs

mrclay opened the pull request:
Fixes #8524

BREAKING CHANGE:
Several internal data structures are no longer stored within the plugin-accessible config space; the removed keys are listed in docs/guides/upgrading.rst. The long-deprecated remove_blacklist() function has also been removed.

ewinslow commented on docs/guides/upgrading.rst:
```diff
@@ -224,14 +224,27 @@ Messages will no longer get the metadata 'msg' for newly created messages. This
+ - ``translations``
+ - ``viewpath``
  - ``views``
+ - ``view_path``
```
Seems like a reasonable set to remove. I don't recall any being on @iionly's list...

mrclay replied:
I figures views has some refactoring left, this clears the path more.

ewinslow commented:
Lgtm. Why `$GLOBALS`?

mrclay commented:
$GLOBALS is unambiguously global, context independent, and ugly (motivation to get rid of it ASAP).

mrclay commented:
Will have to wait a couple days to finish this up unless anyone wants to take it.

ewinslow commented:
@mrclay Ugly is good.

ewinslow commented:
@mrclay What else is there to do? Why not just commit what we have here now?

mrclay committed:
chore(config): Removes several $CONFIG keys and deprecated APIs

Fixes #8524

BREAKING CHANGE:
Several internal data structures are no longer stored within the plugin-
accessible config space; the removed keys are listed in docs/guides/upgrading.rst.
The long-deprecated remove_blacklist() function has also been removed.

mrclay commented:
Should be ready.

ewinslow commented:
Cool! Now to rebase...
